lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -676,6 +676,12 @@ enum class FunctionSynthesisState {
   Done
 };
 
+/// Stores information about `apply` instructions that `PrimalGen` calculates.
+struct ApplyInfo {
```

Comment:
Nit: `ApplyInfo` may be a little too generic. I feel that this is just about the activity (echoing activity analysis) of nested function applications. How about `NestedApplyActivity`?

Reply:
sounds good